omnibus/resources/agent/msi/cal/usercreate.cpp

```diff
@@ -326,15 +370,17 @@ int doesUserExist(MSIHANDLE hInstall, const CustomActionData& data)
     ZeroMemory(refDomain, (cchRefDomain + 1) * sizeof(wchar_t));
 
     // try it again
-    bRet = LookupAccountName(NULL, data.getQualifiedUsername().c_str(), newsid, &cbSid, refDomain, &cchRefDomain, &use);
+    bRet = LookupAccountName(NULL, userToTry, newsid, &cbSid, refDomain, &cchRefDomain, &use);
     if (!bRet) {
         err = GetLastError();
         WcaLog(LOGMSG_STANDARD, "Failed to lookup account name %d", GetLastError());
         retval = -1;
```

Comment:
I'm not sure why we try `LookupAccountName` again here, but in case this call returns `ERROR_NONE_MAPPED` I think we shouldn't fail here: it's the happy case where the user doesn't exist.

Reply:
We try again with the username formatted slightly differently (`.\username` rather than just `username`) to prevent checking the domain controller; it uses the local database only.  This is the core of the whole fix.
It doesn't fail; it's a "semi-bool".  1 for exists (true) 0 for doesn't exist (false), -1 for error.

Reply:
My point is that we never return 0 if we reach this point. Shouldn't we return 0 (doesn't exist) if `err == ERROR_NONE_MAPPED`?

Reply:
I don't understand the question... line 340?

Reply:
I mean to add:
```
if (err == ERROR_NONE_MAPPED) {
     return 0;
}
```
In a similar way as above. Otherwise, after reaching this part of the code, we would return -1 if the user doesn't exist.

Reply:
If this is intended, then ignore my comment.